Actual error is not visible when sending a message

The second time running the example of http://localhost:8080/exist/apps/messaging-replication/messaging-send.html (10 messages) the following error appears:

```
2016-11-04 20:39:56,419 [qtp362432678-39] ERROR (Sender.java [send]:177) - javax.jms.JMSException: Error while attempting to add new Connection to the pool 
2016-11-04 20:39:56,420 [qtp362432678-39] ERROR (SendMessage.java [eval]:90) - exerr:ERROR Error while attempting to add new Connection to the pool 
```

this must be introduced in an attempt to re-use connections,.

actually it was a misconfiguration; a wrong hostname, but the root cause is not visible

```
2016-11-04 21:22:10,611 [qtp362432678-166] INFO  (SenderConnectionFactory.java [getConnectionFactoryInstance]:59) - Creating new connection factory for tcp://miniserver.locall:61616 
2016-11-04 21:22:10,611 [qtp362432678-166] INFO  (SenderConnectionFactory.java [getConnectionFactoryInstance]:66) - Connection factory: org.apache.activemq.pool.PooledConnectionFactory 
2016-11-04 21:22:10,636 [qtp362432678-166] ERROR (Sender.java [send]:177) - Error while attempting to add new Connection to the pool 
javax.jms.JMSException: Error while attempting to add new Connection to the pool
        at org.apache.activemq.jms.pool.PooledConnectionFactory.createJmsException(PooledConnectionFactory.java:266) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.apache.activemq.jms.pool.PooledConnectionFactory.createConnection(PooledConnectionFactory.java:225) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.apache.activemq.jms.pool.PooledConnectionFactory.createConnection(PooledConnectionFactory.java:204) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.exist.jms.send.Sender.send(Sender.java:130) [exist-messaging-replication-0.9.13.jar:?]
        at org.exist.jms.xquery.messaging.SendMessage.eval(SendMessage.java:87) [exist-messaging-replication-0.9.13.jar:?]
        at org.exist.xquery.BasicFunction.eval(BasicFunction.java:73) [exist.jar:?]
        at org.exist.xquery.InternalFunctionCall.eval(InternalFunctionCall.java:41) [exist.jar:?]
        at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58) [exist.jar:?]
        at org.exist.xquery.LetExpr.eval(LetExpr.java:111) [exist.jar:?]
        at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71) [exist.jar:?]
        at org.exist.xquery.ForExpr.processItem(ForExpr.java:239) [exist.jar:?]
        at org.exist.xquery.ForExpr.eval(ForExpr.java:179) [exist.jar:?]
        at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58) [exist.jar:?]
        at org.exist.xquery.LetExpr.eval(LetExpr.java:111) [exist.jar:?]
        at org.exist.xquery.LetExpr.eval(LetExpr.java:111) [exist.jar:?]
        at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71) [exist.jar:?]
        at org.exist.xquery.PathExpr.eval(PathExpr.java:276) [exist.jar:?]
        at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71) [exist.jar:?]
        at org.exist.xquery.XQuery.execute(XQuery.java:253) [exist.jar:?]
        at org.exist.xquery.XQuery.execute(XQuery.java:185) [exist.jar:?]
        at org.exist.http.servlets.XQueryServlet.process(XQueryServlet.java:483) [exist-optional.jar:?]
        at org.exist.http.servlets.XQueryServlet.doPost(XQueryServlet.java:196) [exist-optional.jar:?]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707) [servlet-api-3.1.jar:3.1.0]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [servlet-api-3.1.jar:3.1.0]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:845) [jetty-servlet-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:583) [jetty-servlet-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:566) [jetty-security-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226) [jetty-server-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1174) [jetty-server-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511) [jetty-servlet-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [jetty-server-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1106) [jetty-server-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:159) [jetty-server-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:74) [jetty-server-9.3.9.v20160517.jar:9.3.9.v20160517]
        at org.exist.http.urlrewrite.Forward.doRewrite(Forward.java:50) [exist-optional.jar:?]
        at org.exist.http.urlrewrite.XQueryURLRewrite.doRewrite(XQueryURLRewrite.java:545) [exist-optional.jar:?]
....
Caused by: javax.jms.JMSException: Could not connect to broker URL: tcp://miniserver.locall:61616. Reason: java.net.UnknownHostException: miniserver.locall
        at org.apache.activemq.util.JMSExceptionSupport.create(JMSExceptionSupport.java:36) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:374) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:304) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.ActiveMQConnectionFactory.createConnection(ActiveMQConnectionFactory.java:244) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.jms.pool.PooledConnectionFactory.createConnection(PooledConnectionFactory.java:275) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.apache.activemq.jms.pool.PooledConnectionFactory$1.makeObject(PooledConnectionFactory.java:95) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.apache.activemq.jms.pool.PooledConnectionFactory$1.makeObject(PooledConnectionFactory.java:92) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.apache.commons.pool2.impl.GenericKeyedObjectPool.create(GenericKeyedObjectPool.java:1041) ~[commons-pool2-2.4.2.jar:2.4.2]
        at org.apache.commons.pool2.impl.GenericKeyedObjectPool.addObject(GenericKeyedObjectPool.java:1221) ~[commons-pool2-2.4.2.jar:2.4.2]
        at org.apache.activemq.jms.pool.PooledConnectionFactory.createConnection(PooledConnectionFactory.java:221) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        ... 67 more
Caused by: java.net.UnknownHostException: miniserver.locall
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184) ~[?:1.8.0_111]
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[?:1.8.0_111]
        at java.net.Socket.connect(Socket.java:589) ~[?:1.8.0_111]
        at org.apache.activemq.transport.tcp.TcpTransport.connect(TcpTransport.java:525) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.transport.tcp.TcpTransport.doStart(TcpTransport.java:488) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.util.ServiceSupport.start(ServiceSupport.java:55) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.transport.AbstractInactivityMonitor.start(AbstractInactivityMonitor.java:168) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.transport.InactivityMonitor.start(InactivityMonitor.java:52) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:64) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.transport.WireFormatNegotiator.start(WireFormatNegotiator.java:72) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:64) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:64) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:354) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:304) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.ActiveMQConnectionFactory.createConnection(ActiveMQConnectionFactory.java:244) ~[activemq-client-5.14.1.jar:5.14.1]
        at org.apache.activemq.jms.pool.PooledConnectionFactory.createConnection(PooledConnectionFactory.java:275) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.apache.activemq.jms.pool.PooledConnectionFactory$1.makeObject(PooledConnectionFactory.java:95) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.apache.activemq.jms.pool.PooledConnectionFactory$1.makeObject(PooledConnectionFactory.java:92) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        at org.apache.commons.pool2.impl.GenericKeyedObjectPool.create(GenericKeyedObjectPool.java:1041) ~[commons-pool2-2.4.2.jar:2.4.2]
        at org.apache.commons.pool2.impl.GenericKeyedObjectPool.addObject(GenericKeyedObjectPool.java:1221) ~[commons-pool2-2.4.2.jar:2.4.2]
        at org.apache.activemq.jms.pool.PooledConnectionFactory.createConnection(PooledConnectionFactory.java:221) ~[activemq-jms-pool-5.14.1.jar:5.14.1]
        ... 67 more
2016-11-04 21:22:10,638 [qtp362432678-166] ERROR (SendMessage.java [eval]:90) - exerr:ERROR Error while attempting to add new Connection to the pool 
```

--- a/build.properties
+++ b/build.properties
@@ -3,7 +3,7 @@ exist.dir=../exist/
 
 # Extension details
 module.name=messaging-replication
-module.version=0.9.13
+module.version=0.9.14
 module.title=Messaging and Replication
 
 # Build instructions

--- a/build.xml
+++ b/build.xml
@@ -154,11 +154,9 @@
         </zip>
     </target>
 
-    <target name="install" description="Install jar files into ${exist.dir}/lib/extensions">
-        <copy file="${build.dir}/exist-${module.name}-${module.version}.jar" todir="${exist.dir}/lib/user"/>
-
-        <copy todir="${exist.dir}/lib/user">
-            <fileset dir="${java.libs}" includes="*.jar" excludes="*sources*,*javadoc*"/>
+    <target name="install" description="Install jar files into eXist autodeploy directory">
+        <copy todir="${exist.dir}/autodeploy">
+            <fileset dir="${dist.dir}" includes="*.xar"/>
         </copy>
     </target>
 

--- a/build.xml
+++ b/build.xml
@@ -138,7 +138,7 @@
         <zip destfile="${dist.dir}/${module.name}-ui-${module.version}.xar">
             <fileset dir="${web.dir}"/>
             <zipfileset dir="${build.config}/application" includes="*.xml"/>
-            <zipfileset dir="${build.config}/library" includes="*.xql"/>
+            <zipfileset dir="${build.config}/application" includes="*.xql"/>
         </zip>
 
         <!-- create library -->

--- a/build.xml
+++ b/build.xml
@@ -138,6 +138,7 @@
         <zip destfile="${dist.dir}/${module.name}-ui-${module.version}.xar">
             <fileset dir="${web.dir}"/>
             <zipfileset dir="${build.config}/application" includes="*.xml"/>
+            <zipfileset dir="${build.config}/library" includes="*.xql"/>
         </zip>
 
         <!-- create library -->
@@ -149,6 +150,7 @@
                 <exclude name="*-sources.jar"/>
             </zipfileset>
             <zipfileset dir="${build.config}/library" includes="*.xml"/>
+            <zipfileset dir="${build.config}/library" includes="*.xql"/>
         </zip>
     </target>
 

--- a/config/application/post-install.xql
+++ b/config/application/post-install.xql
@@ -1,0 +1,14 @@
+xquery version "3.0";
+
+import module namespace xdb="http://exist-db.org/xquery/xmldb";
+
+(: The following external variables are set by the repo:deploy function :)
+
+(: file path pointing to the exist installation directory :)
+declare variable $home external;
+(: path to the directory containing the unpacked .xar package :)
+declare variable $dir external;
+(: the target collection into which the app is deployed :)
+declare variable $target external;
+
+xdb:create-collection($target || "/demo/replication", "replicated")

--- a/config/application/pre-install.xql
+++ b/config/application/pre-install.xql
@@ -29,4 +29,5 @@ declare function local:mkcol($collection, $path) {
 
 (: store the collection configuration :)
 local:mkcol("/db/system/config", $target),
-xdb:store-files-from-pattern(concat("/system/config", $target), $dir, "*.xconf")
+xdb:store-files-from-pattern(concat("/system/config", $target), $dir, "*.xconf"),
+if(sm:find-groups-by-groupname("jms")="jms") then () else sm:create-group("jms", "JMS messaging/replication extension")

--- a/config/application/pre-install.xql
+++ b/config/application/pre-install.xql
@@ -1,4 +1,4 @@
-xquery version "1.0";
+xquery version "3.0";
 
 import module namespace xdb="http://exist-db.org/xquery/xmldb";
 

--- a/config/application/repo.xml
+++ b/config/application/repo.xml
@@ -9,5 +9,5 @@
     <type>application</type>
     <target>messaging-replication</target>
     <prepare>pre-install.xql</prepare>
-    <finish/>
+    <finish>post-install.xql</finish>
 </meta>

--- a/config/library/pre-install.xql
+++ b/config/library/pre-install.xql
@@ -1,0 +1,33 @@
+xquery version "1.0";
+
+import module namespace xdb="http://exist-db.org/xquery/xmldb";
+
+(: The following external variables are set by the repo:deploy function :)
+
+(: file path pointing to the exist installation directory :)
+declare variable $home external;
+(: path to the directory containing the unpacked .xar package :)
+declare variable $dir external;
+(: the target collection into which the app is deployed :)
+declare variable $target external;
+
+declare function local:mkcol-recursive($collection, $components) {
+    if (exists($components)) then
+        let $newColl := concat($collection, "/", $components[1])
+        return (
+            xdb:create-collection($collection, $components[1]),
+            local:mkcol-recursive($newColl, subsequence($components, 2))
+        )
+    else
+        ()
+};
+
+(: Helper function to recursively create a collection hierarchy. :)
+declare function local:mkcol($collection, $path) {
+    local:mkcol-recursive($collection, tokenize($path, "/"))
+};
+
+(: store the collection configuration :)
+local:mkcol("/db/system/config", $target),
+xdb:store-files-from-pattern(concat("/system/config", $target), $dir, "*.xconf"),
+if(sm:find-groups-by-groupname("jms")="jms") then () else sm:create-group("jms", "JMS messaging/replication extension")

--- a/config/library/pre-install.xql
+++ b/config/library/pre-install.xql
@@ -1,33 +1,3 @@
-xquery version "1.0";
+xquery version "3.0";
 
-import module namespace xdb="http://exist-db.org/xquery/xmldb";
-
-(: The following external variables are set by the repo:deploy function :)
-
-(: file path pointing to the exist installation directory :)
-declare variable $home external;
-(: path to the directory containing the unpacked .xar package :)
-declare variable $dir external;
-(: the target collection into which the app is deployed :)
-declare variable $target external;
-
-declare function local:mkcol-recursive($collection, $components) {
-    if (exists($components)) then
-        let $newColl := concat($collection, "/", $components[1])
-        return (
-            xdb:create-collection($collection, $components[1]),
-            local:mkcol-recursive($newColl, subsequence($components, 2))
-        )
-    else
-        ()
-};
-
-(: Helper function to recursively create a collection hierarchy. :)
-declare function local:mkcol($collection, $path) {
-    local:mkcol-recursive($collection, tokenize($path, "/"))
-};
-
-(: store the collection configuration :)
-local:mkcol("/db/system/config", $target),
-xdb:store-files-from-pattern(concat("/system/config", $target), $dir, "*.xconf"),
 if(sm:find-groups-by-groupname("jms")="jms") then () else sm:create-group("jms", "JMS messaging/replication extension")

--- a/config/library/repo.xml
+++ b/config/library/repo.xml
@@ -7,4 +7,5 @@
     <license>GNU-LGPL</license>
     <copyright>true</copyright>
     <type>library</type>
+    <prepare>pre-install.xql</prepare>
 </meta>

--- a/java/src/org/exist/jms/replication/shared/ClientParameters.java
+++ b/java/src/org/exist/jms/replication/shared/ClientParameters.java
@@ -49,7 +49,7 @@ public abstract class ClientParameters {
     protected String connectionUsername = null;
     protected String connectionPassword = null;
 
-    protected Properties props = new Properties();
+    protected final Properties props = new Properties();
 
     /**
      * Retrieve configuration value when available as String.
@@ -81,7 +81,7 @@ public abstract class ClientParameters {
     public void setMultiValueParameters(final Map<String, List<?>> params) {
 
         // Iterate over parameters
-        params.entrySet().stream().forEach((entry) -> {
+        params.entrySet().forEach((entry) -> {
             // Get key, values
             final String key = entry.getKey();
             final List<?> values = entry.getValue();
@@ -103,7 +103,7 @@ public abstract class ClientParameters {
      */
     public void setSingleValueParameters(final Map<String, List<?>> params) {
 
-        params.keySet().stream().forEach((key) -> {
+        params.keySet().forEach((key) -> {
             final String value = getConfigurationValue(params, key);
             if (value != null) {
                 props.setProperty(key, value);

--- a/java/src/org/exist/jms/send/Sender.java
+++ b/java/src/org/exist/jms/send/Sender.java
@@ -173,6 +173,18 @@ public class Sender {
             // Return report
             return createReport(message, producer, jmsConfig);
 
+        } catch (final JMSException ex) {
+            LOG.error(ex.getMessage(), ex);
+
+            Throwable cause = ex.getCause();
+
+            if("Error while attempting to add new Connection to the pool".contentEquals(ex.getMessage()) && cause!=null){
+                throw new XPathException(cause.getMessage());
+
+            } else {
+                throw new XPathException(ex.getMessage());
+            }
+
         } catch (final Throwable ex) {
             LOG.error(ex.getMessage(), ex);
             throw new XPathException(ex.getMessage());

--- a/java/src/org/exist/jms/shared/Report.java
+++ b/java/src/org/exist/jms/shared/Report.java
@@ -135,7 +135,7 @@ public class Report {
      */
     public List<String> getErrorMessages() {
         final List<String> errorMessages = new ArrayList<>();
-        errors.stream().forEach((t) -> errorMessages.add(t.getMessage()));
+        errors.forEach((t) -> errorMessages.add(t.getMessage()));
         return errorMessages;
     }
 
@@ -173,7 +173,7 @@ public class Report {
 
         final List<ReportItem> listenerErrors = getReportItems();
         if (!listenerErrors.isEmpty()) {
-            listenerErrors.stream().forEach((ri) -> ri.writeError(builder));
+            listenerErrors.forEach((ri) -> ri.writeError(builder));
         }
 
         builder.endElement();

--- a/java/src/org/exist/jms/shared/eXistMessage.java
+++ b/java/src/org/exist/jms/shared/eXistMessage.java
@@ -174,7 +174,7 @@ public class eXistMessage {
         if (!keys.isEmpty()) {
             sb.append("###  ");
 
-            keys.stream().forEach((key) -> {
+            keys.forEach((key) -> {
                 final Object val = metaData.get(key);
                 if (val != null) {
                     sb.append(key).append("='").append(val.toString()).append("'  ");

--- a/java/src/org/exist/jms/xquery/management/ListReceivers.java
+++ b/java/src/org/exist/jms/xquery/management/ListReceivers.java
@@ -69,7 +69,7 @@ public class ListReceivers extends BasicFunction {
         final ValueSequence returnSequence = new ValueSequence();
 
         // Collect IDs
-        manager.getIds().stream().forEach((id) -> returnSequence.add(new IntegerValue(id)));
+        manager.getIds().forEach((id) -> returnSequence.add(new IntegerValue(id)));
 
         // Return IDs
         return returnSequence;

--- a/web/demo/replication/collection.xconf
+++ b/web/demo/replication/collection.xconf
@@ -1,0 +1,49 @@
+<collection xmlns="http://exist-db.org/collection-config/1.0">
+
+    <!--
+        To enable replication for the 'replication' collection and all child collections:
+
+        -1- Open this document in eXide
+
+        -2- Click "Save"
+
+        Popup appears "You have saved a collection configuration file. Would you like to
+        apply it to collection /db/apps/messaging-replication/demo/replication now?"
+
+        -3- Click "OK"
+
+        The collection.xconf document is now installed as
+            /db/system/config/db/apps/messaging-replication/demo/replication/collection.xconf
+
+    -->
+
+
+    <triggers>
+        <trigger class="org.exist.jms.replication.publish.ReplicationTrigger">
+            <!--
+                Class name of the initial context provider, default value
+                for ActiveMQ
+                see javax.naming.Context#INITIAL_CONTEXT_FACTORY
+            -->
+            <parameter name="java.naming.factory.initial"
+                       value="org.apache.activemq.jndi.ActiveMQInitialContextFactory"/>
+            <!--
+                URL of the message broker, default value for ActiveMQ
+                see javax.naming.Context#PROVIDER_URL
+            -->
+            <parameter name="java.naming.provider.url"
+                       value="tcp://localhost:61616"/>
+            <!--
+                Lookup connection factory
+                see javax.naming.InitialContext#lookup(String)
+            -->
+            <parameter name="connection-factory" value="ConnectionFactory"/>
+            <!--
+                Lookup destination (topic)
+                see javax.naming.InitialContext#lookup(String)
+            -->
+            <parameter name="destination"
+                       value="dynamicTopics/eXistdb-replication-demo"/>
+        </trigger>
+    </triggers>
+</collection>

--- a/web/demo/replication/createDocuments.xq
+++ b/web/demo/replication/createDocuments.xq
@@ -1,0 +1,14 @@
+xquery version "3.1";
+
+(:~
+: Simple script to create a small number of resources to test replication of creation of the resources.
+:)
+
+import module namespace config = "http://exist-db.org/extension/jms/config" at "../../modules/config.xqm";
+
+let $root := $config:app-root || "/demo/replication/replicated"
+
+for $i in (1000 to 1010)
+return
+    xmldb:store($root, "mydoc" || $i || ".xml", <doc>{ util:uuid() }</doc>)
+

--- a/web/demo/replication/removeDocuments.xq
+++ b/web/demo/replication/removeDocuments.xq
@@ -1,0 +1,13 @@
+xquery version "3.1";
+
+(:~
+: Simple script to remove resources to test the replication of deletion the resources.
+:)
+
+import module namespace config = "http://exist-db.org/extension/jms/config" at "../../modules/config.xqm";
+
+let $root := $config:app-root || "/demo/replication/replicated"
+
+for $document in xmldb:get-child-resources($root)
+return
+    xmldb:remove($root, $document)

--- a/web/demo/replication/startListener.xq
+++ b/web/demo/replication/startListener.xq
@@ -1,7 +1,14 @@
 xquery version "3.1";
 
 (:~
-: Simple script to start the replication listener
+: Simple script to start the replication listener.
+:
+: To have this script executed when eXist-db is started:
+:
+: - enable in conf.xml the XQueryStartupTrigger
+: - carefully read the instructions about permissons
+: - copy this script into /db/system/autostart
+: - restart eXist-db, check logs for messages from the XQueryStartupTrigger
 :)
 
 import module namespace replication="http://exist-db.org/xquery/replication" at "java:org.exist.jms.xquery.ReplicationModule";

--- a/web/demo/replication/startListener.xq
+++ b/web/demo/replication/startListener.xq
@@ -1,0 +1,19 @@
+xquery version "3.1";
+
+(:~
+: Simple script to start the replication listener
+:)
+
+import module namespace replication="http://exist-db.org/xquery/replication" at "java:org.exist.jms.xquery.ReplicationModule";
+
+let $jmsConfiguration := map {
+        "java.naming.factory.initial" := "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        "java.naming.provider.url" := "tcp://localhost:61616",
+        "connection-factory" := "ConnectionFactory",
+        "destination" := "dynamicTopics/eXistdb-replication-example",
+        "subscriber.name" := "SubscriptionId",
+        "connection.client-id" := "ClientId"
+    }
+
+return
+    replication:register($jmsConfiguration)

--- a/web/templates/page.html
+++ b/web/templates/page.html
@@ -53,7 +53,7 @@
                                         </li>
                                     </ul>
                                 </li>
-                                <li class="dropdown" id="functions">
+                                <li class="dropdown" id="messaging">
                                     <a href="#" class="dropdown-toggle" data-toggle="dropdown">Messaging</a>
                                     <ul class="dropdown-menu">
                                         <li>
@@ -70,7 +70,7 @@
                                         </li>
                                     </ul>
                                 </li>
-                                <li class="dropdown" id="functions">
+                                <li class="dropdown" id="replication">
                                     <a href="#" class="dropdown-toggle" data-toggle="dropdown">Replication</a>
                                     <ul class="dropdown-menu">
                                         <li>

